djstripe/views.py

```diff
@@ -261,19 +273,23 @@ def form_valid(self, form):
         subscription = customer.subscription.cancel()
 
         if subscription.status == subscription.STATUS_CANCELED:
-            # If no pro-rate, they get kicked right out.
-            messages.info(self.request, "Your subscription is now cancelled.")
-            # logout the user
-            auth_logout(self.request)
-            return redirect("home")
+            return self.status_cancel()
         else:
             # If pro-rate, they get some time to stay.
-            messages.info(self.request, "Your subscription status is now '{status}' until '{period_end}'".format(
+            messages.info(self.request, self.subscription_status_msg.format(
                 status=subscription.status, period_end=subscription.current_period_end)
             )
 
         return super(CancelSubscriptionView, self).form_valid(form)
 
+    def status_cancel(self):
```

Comment:
I split it because logging the user out is not functionality you necessarily want (in fact, unless membership is specifically premium-only, why would you want that?). Short of adding a setting (which seems overkill), you can override status_cancel to be a noop.